src/timers/CTimers.cpp

```diff
@@ -0,0 +1,152 @@
+/// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2023 The MMapper Authors
+
+#include "CTimers.h"
+
+#include <QMutexLocker>
+#include <QSettings>
+#include <QTimer>
+
+CTimers::CTimers(QObject *parent)
+    : QObject(parent)
+{
+    m_nextId = 1;
+}
+
+CTimers::~CTimers() {}
+
+void CTimers::addTimer(QByteArray name, QByteArray desc)
+{
+    QMutexLocker locker(&m_lock);
+
+    TTimer timer;
+
+    timer.name = name;
+    timer.desc = desc;
+    timer.timer.start();
+    timer.id = m_nextId++;
+
+    m_timers.push_back(timer);
+}
+
+bool CTimers::removeCountdown(QByteArray name)
+{
+    QMutexLocker locker(&m_lock);
+
+    for (std::list<TTimer>::iterator it = m_countdowns.begin(); it != m_countdowns.end(); ++it) {
+        if (it->name == name) {
+            m_countdowns.erase(it);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+bool CTimers::removeTimer(QByteArray name)
+{
+    QMutexLocker locker(&m_lock);
+
+    for (std::list<TTimer>::iterator it = m_timers.begin(); it != m_timers.end(); ++it) {
+        if (it->name == name) {
+            m_timers.erase(it);
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void CTimers::addCountdown(QByteArray name, QByteArray desc, int time)
+{
+    QMutexLocker locker(&m_lock);
+
+    TTimer countdown;
+    countdown.name = name;
+    countdown.desc = desc;
+    countdown.timer.start();
+    countdown.id = m_nextId++;
+    countdown.duration = time;
+
+    // append the shot-down event pending
+    QTimer::singleShot(time, this, SLOT(finishCountdownTimer()));
+
+    m_countdowns.push_back(countdown);
+}
+
+void CTimers::finishCountdownTimer()
+{
+    QMutexLocker locker(&m_lock);
```

Comment:
I'm wondering if a mutex is even necessary here. Isn't the Thread owned by the same object as the Proxy and Parser so it's the same thread?

Reply:
Well, I wasn't sure. 
At least 3 things may happen (in theory) at the same time:
- timer for the countdown kicking in and tweaking the countdowns data
- User input adds new entries or remove the old ones
- Proxy (if i recall that right) attempting to retrieve the entries

Now, I'm not sure if this is really in the same thread or not, so I kept the same protection as the old code in PandoraMapper.

Reply:
I left it